### PR TITLE
Fix: No error message when entering a invalid email

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

No error message when entering a invalid email for a new chat
### Fixed Issues

$ #16884 
PROPOSAL: [#16884 (comment)](https://github.com/Expensify/App/issues/16884#issuecomment-1502149967)

### Tests

1. Login with any account
2. Click on the green plus icon and choose New chat
3. Enter a invalid email (i.e new1:1@test.com)
4. Verify "Invalid email" error message is displayed
5. Switch to Spanish and verify 'Email inválido' is displayed
- [x] Verify that no errors appear in the JS console

### Offline tests
Same above

### QA Steps

1. Login with any account
2. Click on the green plus icon and choose New chat
3. Enter a invalid email (i.e new1:1@test.com)
4. Verify "Invalid email" error message is displayed
5. Switch to Spanish and verify 'Email inválido' is displayed
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 11-04-2023 15:00:12.webm](https://user-images.githubusercontent.com/129500732/231248182-d0965856-d11f-41ab-9b3b-5c04c9163ef0.webm)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/129500732/231129413-328decc8-be81-4117-978a-3ca02ac0a901.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-04-12 at 00 53 01

</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-04-12 at 00 51 29

</details>

<details>
<summary>iOS</summary>
Screenshot 2023-04-12 at 01 32 13

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/129500732/231130367-8fb844d5-564e-4d59-928a-fb6c33a9a4f3.mp4

</details>

### CODE REVIEW

**Comment by dukenv0307** [comment]
@Beamanator This is my first PR, please help to take a look at it and enable 10 workflows awaiting approval

File: src/languages/en.js
Comment (Beamanator) [reply src/languages/en.js:101]: @fedirjh just wondering, why do you prefer having the error message inside `messages` instead of here? I feel like here makes more sense since this is a generic set of errors

Comment (dukenv0307) [reply src/languages/en.js:101]: @fedirjh Do you want the `invalidEmail` message similar to `errorMessageInvalidPhone` inside `messages` or leave it as it is?

**Comment by Beamanator** [comment]
@neil-marcellini assigning you since it seems you were assigned on the original issue (thanks for pointing that out @fedirjh )

File: src/languages/en.js
Comment (neil-marcellini) [reply src/languages/en.js:101]: I would put it inside `messages` because it's more of an informative message rather than an error message, but it's not a big deal either way.

File: src/libs/OptionsListUtils.js
Comment (neil-marcellini) [inline src/libs/OptionsListUtils.js:807]: Maybe we could add a default, for the case where it's not clearly a phone number or email, but it's invalid for both?

For example when I enter `new1:1` it should say "Invalid email or phone number".

**Review by neil-marcellini — COMMENTED** [review]
Look good overall. I left a few small comments. We're also waiting for copy. I'll review again once we have the copy locked down.

File: src/libs/OptionsListUtils.js
Comment (dukenv0307) [reply src/libs/OptionsListUtils.js:807]: 1. I think that when user has entered the character as a letter that means they want to search by name or email
2. If you want do add this case, can you defind when `searchValue` is not clearly a phone number or email? When it has both letters and numbers, but doesnt't have '@' ?

**Comment by dukenv0307** [comment]
@Beamanator Should we hold this PR until the correct copy is confirmed

**Comment by dukenv0307** [comment]
@neil-marcellini I saw davidcardoza confirmed that the message of invalid email is "Invalid email". Actually it's set to `Invalid email` in English and `Email inválido` in Spanish
https://github.com/Expensify/App/issues/16884#issuecomment-1510423279
And I moved it into `messages` please check again

**Review by fedirjh — APPROVED** [review]
LGTM based on https://github.com/Expensify/App/issues/16884#issuecomment-1510423279 🚀

cc @neil-marcellini 

🎀👀🎀 C+ reviewed

**Review by neil-marcellini — APPROVED** [review]

**Pull request merged**
